### DISCUSSION TITLE
Passing parametrized method parameter

### DISCUSSION THREAD

**Discussion by opxh** [dotnet/csharplang#1669] [Q&A]
Hi, there!
Just found out the problem with limitations for generic method;
 I'l describe on expample: 

I'm expecting what the call of "Set" for "UIEdit" would pass correclty

```csharp
// here's "Set":
public static IUIEdit<T> Set<T>(this IUIEdit<T> edit, T value)
{ }

// here's "UIEdit":
public interface IUIEdit<T> : IUIControl, IHasValue<T>
{ }

public class UIEdit<T> : UIControl, IUIEdit<T>
{ }

public class UIEdit : UIEdit<string>
{ }
```

But, I'm facing the problem, what actually not this "Set" mapped to it, but this:
```csharp
public static T Set<T>(this T control, params string[] path) where T : IUITreeNode
{ }
```
> "The type UIEdit canot be used as a type parameter 'T' in the generic type or method 'Set<T>(T, params string[]). There is no implicit reference conversion from UIEdit to UITreeNode"

p.s. they are both in one namespace, but both are limited by those interfaces, why is that happening?
I mean, please, dont offer me to split namespaces.
p.s.s. and yeah, I know if I'l change the last "Set" to non-generic - it could be working great.

**What can I do in this situation except the things in p.s?**

**Comment by yaakov-h** [comment]
Do you have your C# language version set to 7.3, or is it lower?

**Comment by opxh** [comment]
I do, it is "C# latest major version (default)" in the select box of "Language version" in project props -> build -> advanced, and the major version in the box is "C# 7.3" and this code 
```csharp
typeof(string).Assembly.ImageRuntimeVersion
```
returned "v4.0.30319"

**Comment by Neme12** [comment]
latest major version = 7.0
latest minor version = 7.3

**Comment by opxh** [comment]
What does "major" means in this context? The greatest number of version, did I get it wrong?

**Comment by Neme12** [comment]
7 is the major version number, 3 is the minor version

**Comment by opxh** [comment]
now I've got same problem with generic UIEdit 

```csharp
 public static IUIEdit<T> Set<T>(this IUIEdit<T> edit, T value)
        {
            edit.Add(new SetToEditCommand(edit, value.ToString()));
            return edit;
        }
```
(it's ok when I'm specifyin signature by UIEdit<T> instead of IUIEdit<T>, but my question is still actual)
